### ISSUE TITLE
Feature request: SARIF Output

### ISSUE THREAD

**Issue by nvuillam** [squizlabs/PHP_CodeSniffer#3496]
Hi, is it in the roadmap to make PHP_CodeSniffer provide [SARIF](https://sarifweb.azurewebsites.net/) output ? (SARIF is the OASIS common format for all analysis tools )

It would help improve its [integration within MegaLinter](https://megalinter.github.io/latest/descriptors/php_phpcs/) :)

Best regards

**Comment by gsherwood** [comment]
No, it's not on the roadmap. I hadn't heard of the format before and I couldn't devote any time to this unless the demand was high. I also wouldn't accept a PR to add the report format to the core unless the demand was there.

**Comment by nvuillam** [comment]
The format is widely used, especially since GitHub reads it to display lint errors directly in Pull Requests UI, so if a good samaritan would submit a PR, you may be wise to accept it :)

As you can see in this table (orange badges), the format is more and more used by many code analyzers, including PHP ones like PHP_Stan ^^

https://megalinter.github.io/v6-alpha/supported-linters/

**Comment by jrfnl** [comment]
Also just looked at the specs and ... jeezs louise.... who's got the time to even read that, let alone implement it ? This would require a study of that standard which could easily take several weeks by the looks of it and does not align with any of the other report formats provided. This won't be straight-forward to implement.

**Comment by jrfnl** [comment]
To be honest, as PHPCS allows for using external reports, in my opinion this is more suited to an external repo providing that report format as I don't see enough demand for this format to justify maintaining it within PHPCS itself.

**Comment by XVilka** [comment]
It's necessary for GitLab adopting SARIF: https://gitlab.com/gitlab-org/gitlab/-/issues/118496

**Comment by soderlind** [comment]
CodeQL, as in GitHub Advanced Security, [also like SARIF](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/uploading-a-sarif-file-to-github)

**Comment by soderlind** [comment]
Typical, found an alternative using [cs2pr](https://github.com/staabm/annotate-pull-request-from-checkstyle?tab=readme-ov-file#annotate-a-pull-request-based-on-a-checkstyle-xml-report)

```yaml
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: '8.3'
    tools: cs2pr, phpcs

- name: Run phpcs
  run: phpcs -q --report=checkstyle src | cs2pr 
```

**Comment by llaville** [comment]
@nvuillam I've a good new for you. As recommanded by @jrfnl, I've started to write a single repository with only the SARIF report support for PHP CodeSniffer 3.3 or higher.

As I did in past with PHPStan (see my old request : https://github.com/phpstan/phpstan/issues/5973) that was not accepted and lead to an alternative solution https://github.com/jbelien/phpstan-sarif-formatter,  my PHP_CodeSniffer SARIF support is based on my package (https://github.com/llaville/sarif-php-sdk) that implement the full specification of SARIF 2.1.0

My first attempt is already pretty good, but I need more days to polish code before to propose a new package to Community.

**Comment by nvuillam** [comment]
This will be a great addition yo MegaLinter :)

**Comment by llaville** [comment]
@nvuillam Integration with MegaLinter will continue on its own thread (just opened 3515)

**Comment by llaville** [comment]
For all, the solution exists now with https://github.com/llaville/sarif-php-sdk/releases/tag/1.2.0
Demo and explains may be found on https://github.com/llaville/sarif-php-sdk/tree/1.2/examples/converters/phpcs
